homeassistant/bootstrap.py

```diff
@@ -380,38 +385,44 @@ def _get_domains(hass: core.HomeAssistant, config: dict[str, Any]) -> set[str]:
     return domains
 
 
-async def _async_log_pending_setups(
-    hass: core.HomeAssistant, domains: set[str], setup_started: dict[str, datetime]
-) -> None:
+async def _async_watch_pending_setups(hass: core.HomeAssistant) -> None:
     """Periodic log of setups that are pending for longer than LOG_SLOW_STARTUP_INTERVAL."""
+    loop_count = 0
+    setup_started: dict[str, datetime] = hass.data[DATA_SETUP_STARTED]
     while True:
-        await asyncio.sleep(LOG_SLOW_STARTUP_INTERVAL)
-        remaining = [domain for domain in domains if domain in setup_started]
+        now = dt_util.utcnow()
+        remaining = setup_started.keys()
```

Comment:
Maybe I'm wrong, but I think that you can just use `setup_started` and drop `remaining`.

You can quickly test with this change:

```suggestion
        remaining = setup_started
```

Reply:
Right, I think I can just drop `remaining` here.